lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2204,7 +2207,10 @@ namespace {
       }
 
       if (Impl.SwiftContext.LangOpts.hasFeature(Feature::NonescapableTypes) &&
-          importer::hasNonEscapableAttr(decl)) {
+          (importer::hasNonEscapableAttr(decl) ||
```

Comment:
Hmm, is this logic adding a non-escapable attribute to the typedef, or its underlying record?

Reply:
The syntax `using VecOfPtr SWIFT_NONESCAPABLE = std::vector<int*>;` will apply the attribute to the type alias on the Clang side, but on the Swift side we add the attribute to the type we create based on the underlying record.

Reply:
Let's assume a Swift target imports two C++ modules:
```cpp
// A.hpp
using AVector SWIFT_NONESCAPABLE = std::vector<int*>;
inline AVector getSomethingNonEscapable();
```
```cpp
// B.hpp
using BVector SWIFT_ESCAPABLE = std::vector<int*>;
inline BVector getSomethingEscapable();
```

How would `AVector` and `BVector` be imported? Would they be the same type?

Reply:
This is a really good point. Unfortunately, they need to have the same underlying type, and my guess is that the escapability would depend on which type alias was imported first. 

Since the Swift compiler needs to attach the escapability information to the type (not the alias), I think the only way around this would be to import the same type twice if the escapability does not match. Do you want me to give that a try in this PR or is it OK in a follow-up?